core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -984,6 +986,8 @@ private static Expression implementNullSemantics(
         : call.getOperands();
     switch (nullAs) {
     case NULL:
+    case IS_NULL:
```

Comment:
Reason of the change:
`IS_NULL`/`IS_NOT_NULL` define the boolean return based on the nullability of original return value. The use of them should not break trivial null handling for operands.